test/test-lib/fixtures.ts

```diff
@@ -1,7 +1,148 @@
+import { sbvrUtils } from '@resin/pinejs';
 import * as Bluebird from 'bluebird';
+import * as fs from 'fs';
 import * as _ from 'lodash';
+import * as path from 'path';
+
+import { Tx } from '@resin/pinejs/out/database-layer/db';
+import { Headers } from 'request';
+import { API_HOST } from '../../src/lib/config';
+import { requestAsync } from '../../src/lib/request';
+
+const { api, root } = sbvrUtils;
+
+interface ApiRequestOpts {
+	url: string;
+	resource: string;
+	headers: Headers;
+	method: string;
+	json: boolean;
+	body: AnyObject;
+}
 
 type Fixtures = Dictionary<PromiseLike<Dictionary<PromiseLike<any>>>>;
+type SemiAwaitedFixtures = Dictionary<Dictionary<PromiseLike<any>>>;
+
+type LoaderFunc = (
+	jsonData: AnyObject,
+	fixtures: SemiAwaitedFixtures,
+	tx: Tx,
+) => PromiseLike<any>;
+
+const buildApiRequestOpts = (args: {
+	resource: string;
+	method?: string;
+	body?: AnyObject;
+	user?: { token: string };
+}): ApiRequestOpts => {
+	const { resource, method = 'POST', body = {}, user } = args;
+	const headers: Headers = { 'X-Forwarded-Proto': 'https' };
+
+	if (user != null) {
+		headers.Authorization = `Bearer ${user.token}`;
+	}
+
+	return {
+		url: `http://${API_HOST}/resin/${resource}`,
+		resource,
+		headers,
+		method,
+		json: true,
+		body,
+	};
+};
+
+const logErrorAndThrow = (message: string, ...args: any[]) => {
+	console.error(message, ...args);
+	throw new Error(message);
+};
+
+const createResource = async (requestOpts: ApiRequestOpts) => {
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow(
+			`Failed to create: ${requestOpts.resource}`,
+			response.statusCode,
+			response.body,
+		);
+	}
+
+	return body;
+};
+
+const loaders: Dictionary<LoaderFunc> = {
```

Comment:
I'd like to get rid of this `Dictionary<LoaderFunc>` and instead have the specific keys listed but I haven't worked out a nice way yet

Reply:
Maybe try something like this? https://github.com/balena-io/open-balena-api/blob/master/src/lib/device-state.ts#L63

Regardless, it's definitely worth putting this down as a `FIXME` comment, unless you plan to fix it in this PR.